Fix Create Terminal Dialog Issues

grolu opened the pull request:
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #2170

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       breaking|feature|bugfix|doc|other
- target_group:   user|operator|developer|dependency
-->
```bugfix user
Fixed several issues with Create Terminal Dialog:
- Disable terminal scheduling on non-managed seeds
- Added error handling in case cluster nodes cannot be retrieved
- Moved alerts from scrollable container to fixed position to make them always visible
```

Lappihuan commented:
@grolu i tested this PR and the terminal dialog behaves how it should.
however shortcuts don't behave correctly.

Shortcuts that target `cp` and `shoot` both error with `terminal cannot be hosted on non-managed seed`.
`shoot` at least should work.

Shortcuts targeting `garden` seem to work.

grolu commented:
Hi @Lappihuan 
I fixed some last things:
- Fixed a navigation issue when navigating back from terminal
- Fixed initial CreateTerminalSession dialog rendering
- Improved tooltips

>Shortcuts that target cp and shoot both error with terminal cannot be hosted on non-managed seed.
shoot at least should work.

- Control Plane terminal shortcut targets are now disabled in case seed is not managed
- For Operators, terminal shortcuts with target shoot are also hosted on the seed, therefore they are now also disabled in case the seed is not managed. Of course, they could be scheduled on the shoot. But enabling this for operators would be cumbersome - we would need to show warnings etc. like we do on the create terminal session dialog.

>Shortcuts targeting garden is created but i actually can't connect to the virtual garden, but that seems to be unrelated to this issue.

Probably unrelated, garden terminals for operators are scheduled on a dedicated cluster. Maybe something is wrong with your config. There is also a related issue: https://github.com/gardener/dashboard/issues/1789

grolu commented on frontend/src/components/GTerminalShortcut.vue:
```diff
@@ -176,6 +187,22 @@ export default {
 
       return true
     },
+    disabledCannotScheduleOnSeed () {
```
Refactored as discussed

grolu commented on frontend/src/components/dialogs/GCreateTerminalSessionDialog.vue:
```diff
@@ -225,11 +263,28 @@ export default {
+          this.state.runtime === TargetEnum.SHOOT
+    },
+    createDisabledNoNodes () {
+      if (this.state.runtime !== TargetEnum.SHOOT && this.state.runtime !== TargetEnum.GARDEN) {
```
yes can be removed

grolu commented on frontend/src/composables/useTerminalConfig.js:
```diff
@@ -37,13 +38,13 @@ export function createTerminalConfigComposable () {
 
     state.node = defaultNode
-    if (!defaultNode) {
+    if (!defaultNode && nodes.length) {
```
nodes defaults to empty array

grolu commented on frontend/src/components/dialogs/GCreateTerminalSessionDialog.vue:
```diff
@@ -89,6 +88,39 @@ SPDX-License-Identifier: Apache-2.0
+          type="error"
+          variant="tonal"
+        >
+          <strong>Cannot schedule terminal on <span class="font-family-monospace">{{ shootName }}</span> cluster</strong><br>
+          No worker nodes available in the cluster. Please check the cluster status and try again.
+        </v-alert>
```
validation approach not possible for Garden Terminal target therefore leave as is for now.

grolu commented on frontend/src/composables/useTerminalConfig.js:
```diff
@@ -28,6 +28,7 @@ export function createTerminalConfigComposable () {
       currentNode,
       privilegedMode,
       nodes = [],
+      canScheduleOnSeed = false,
```
state is set via updateState function which is called in case target changes. If default target changes (e.g. because of `canScheduleOnSeed` changes), the state is reinitialized

petersutter commented on frontend/src/components/dialogs/GCreateTerminalSessionDialog.vue:
```diff
@@ -101,7 +101,7 @@ SPDX-License-Identifier: Apache-2.0
         </v-alert>
         <v-alert
-          v-if="showAdminShootTerminalAlert"
+          v-if="isTerminalInUntrustedEnvironment"
```
isAdmin && isTerminalInUntrustedEnvironment

petersutter commented on frontend/src/components/GTerminalShortcut.vue:
```diff
@@ -46,18 +46,22 @@ SPDX-License-Identifier: Apache-2.0
         <template #tooltip>
-          <span v-if="!disabled">
+          <div v-if="isAdmin && !canScheduleOnSeed">
+            Terminal shortcut not available on non-managed seed
```
use same/similar text as on the create cluster dialog `Terminals can only be scheduled if the seed is a managed seed`

petersutter commented on frontend/src/components/dialogs/GCreateTerminalSessionDialog.vue:
```diff
@@ -238,6 +293,12 @@ export default {
       }
     },
+    defaultTarget (value) {
+      this.targetTab.selectedTarget = value
```
default target selection does not work for garden terminal https://localhost:8443/namespace/garden/term/garden#target-tab

petersutter commented on backend/lib/services/shoots.js:
```diff
@@ -304,7 +304,7 @@ exports.info = async function ({ user, namespace, name }) {
   if (shoot.spec.seedName) {
     const seed = getSeed(getSeedNameFromShoot(shoot))
-    if (seed && namespace !== 'garden') {
+    if (seed) {
```
bug not related to this line:
open a cluster terminal for a shoot cluster, that has a non-managed seed. This works for the first terminal to be created, but for the second terminal the error `terminal cannot be hosted on non-managed seed` is shown

grolu replied:
Fixed, missing `updateSettings` in `initialize` function

petersutter reviewed (approved):
/lgtm
